File: src/main/java/com/graqr/threshr/model/redsky/products/FindsPost.java
package com.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record FindsPost(String id, String caption, String url,
                        @JsonProperty("placeholder_color")
                        String placeholderColor,
                        User user,
                        Image image) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/FindsPost.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class FindsPost(
    val id: String,
    val caption: String,
    val url: String,
    @param:JsonProperty("placeholder_color") val placeholderColor: String,
    val user: User,
    val image: Image
) {}
